src/conf.c

```diff
@@ -2309,6 +2309,47 @@ int snd_config_make(snd_config_t **config, const char *id,
 	return _snd_config_make(config, &id1, type);
 }
 
+/**
+ * \brief Creates a configuration node and add it to the parent node.
+ * \param[out] config The function puts the handle to the new node at
+ *                    the address specified by \a config.
+ * \param[in] id The id of the new node.
+ * \param[in] type The type of the new node.
+ * \param[in] parent handle for the parent node.
+ * \return Zero if successful, otherwise a negative error code.
+ *
+ * This functions creates a new node of the specified type.
+ * The new node has id \a id, which may be \c NULL.
+ * and adds it to the parent node.
+ *
+ * The value of the new node is zero (for numbers), or \c NULL (for
+ * strings and pointers), or empty (for compound nodes).
+ *
+ * \par Errors:
+ * <dl>
+ * <dt>-ENOMEM<dd>Out of memory.
+ * </dl>
+ */
+int snd_config_make_add(snd_config_t **config, char *id,
```

Comment:
Can be replaced with snd_config_make() + snd_config_add() calls. I don't think that we need to export another fcn for this functionality.